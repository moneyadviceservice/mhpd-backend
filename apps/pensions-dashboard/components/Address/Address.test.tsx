import React from 'react';
import { render } from '@testing-library/react';
import { Address } from '.';

const mockAddress = {
  postalName: 'The House',
  line1: 'Line 1',
  postcode: 'TN1 1AA',
  countryCode: 'GB',
};

const mockFullAddress = {
  ...mockAddress,
  line2: 'Line 2',
  line3: 'Line 3',
  line4: 'Line 4',
  line5: 'Line 5',
};

describe('Address', () => {
  it('renders minimal address correctly', () => {
    const { container } = render(<Address address={mockAddress} />);
    expect(container).toMatchSnapshot();
  });

  it('renders full address correctly', () => {
    const { container } = render(<Address address={mockFullAddress} />);
    expect(container).toMatchSnapshot();
  });
});
