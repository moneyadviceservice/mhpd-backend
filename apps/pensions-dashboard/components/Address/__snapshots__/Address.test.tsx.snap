// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address renders full address correctly 1`] = `
<div>
  <p>
    The House
  </p>
  <p>
    Line 1
  </p>
  <p>
    Line 2
  </p>
  <p>
    Line 3
  </p>
  <p>
    Line 4
  </p>
  <p>
    Line 5
  </p>
  <p>
    TN1 1AA
  </p>
  <p>
    United Kingdom
  </p>
</div>
`;

exports[`Address renders minimal address correctly 1`] = `
<div>
  <p>
    The House
  </p>
  <p>
    Line 1
  </p>
  <p>
    TN1 1AA
  </p>
  <p>
    United Kingdom
  </p>
</div>
`;
